js/custom_event.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types";
+import * as event from "./event";
+import { getPrivateValue } from "./util";
+
+// WeakMaps are recommended for private attributes (see MDN link below)
+// tslint:disable-next-line:max-line-length
+// https://developer.mozilla.org/en-US/docs/Archive/Add-ons/Add-on_SDK/Guides/Contributor_s_Guide/Private_Properties#Using_WeakMaps
+export const customEventAttributes = new WeakMap();
+
+export class CustomEventInit extends event.EventInit
+  implements domTypes.CustomEventInit {
+  detail = null;
```

Comment:
Probably `detail: any | null = null`?

Reply:
👍 but `any | null` would be just `any`.

Reply:
Also, you are double initialising this... I would simply do:

```
detail: any;
```

Here and you have a default argument in the constructor which is then assigning to detail on construction.  Also, I am not sure, but is detail `readonly`?